### ISSUE TITLE
static-challenge does not work on OpenVPN3 

### ISSUE THREAD

**Issue by telboon** [OpenVPN/openvpn3-linux#75]
I am using OpenVPN 3 v15_beta on Ubuntu 20.04.

With my configuration, the `static-challenge` does not seem to trigger when I attempt to login. The expected behavior should be that it should work similar to OpenVPN2 where username, password, and OTP will be asked before connecting.

Is `static-challenge` supported by OpenVPN3? Is this feature taken out by design, is this a bug, or is this a user problem?

OpenVPN version
```
$ openvpn3 version
OpenVPN 3/Linux v15_beta (openvpn3)
OpenVPN core 3.git:HEAD:fce979ec linux x86_64 64-bit
Copyright (C) 2012-2020 OpenVPN Inc. All rights reserved.
```

Configuration with redacted info
```
client
tls-client
remote-cert-tls server
proto udp
remote vpn.example.com 1194 udp
dev tun
topology subnet
auth-user-pass
static-challenge "OTP: " 1
reneg-sec 0
pull

<ca>
</ca>
<cert>
</cert>
<key>
</key>
<tls-crypt>
</tls-crypt>
```

Output of OpenVPN3
```
$ openvpn2 client.ovpn
Credentials needed
Auth User name: user
Auth Password: 
Press CTRL-C to stop the connection

2021-09-23 16:46:04.492981 [STATUS] (StatusMajor.CONNECTION, StatusMinor.CFG_OK) config_path=/net/openvpn/v3/configuration/400f6c55x1f75x4493xbdfcxb3e05777a66d
2021-09-23 16:46:04.493039 [STATUS] (StatusMajor.CONNECTION, StatusMinor.CONN_CONNECTING) 
2021-09-23 16:46:04.505174 [STATUS] (StatusMajor.CONNECTION, StatusMinor.CONN_CONNECTING) 
2021-09-23 16:46:04.534476 [STATUS] (StatusMajor.CONNECTION, StatusMinor.CONN_AUTH_FAILED) Authentication failed

Session manager initiated disconnect...
Disconnected


$ openvpn3 session-start -c client.ovpn
Using configuration profile from file: client.ovpn
Session path: /net/openvpn/v3/sessions/bbc30e60s4a88s4f8bs88b4sfe7373e443df
Auth User name: user
Auth Password: 
session-start: ** ERROR ** Failed to connect: Connection, Client authentication failed: Authentication failed

```

Output of OpenVPN2
```
$ sudo openvpn client.ovpn
Thu Sep 23 16:46:45 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Apr 27 2021
Thu Sep 23 16:46:45 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Enter Auth Username: user
Enter Auth Password: *    
CHALLENGE: OTP:  123456
Thu Sep 23 16:47:07 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]server:1194
Thu Sep 23 16:47:07 2021 UDP link local (bound): [AF_INET][undef]:1194
Thu Sep 23 16:47:07 2021 UDP link remote: [AF_INET]server:1194
Thu Sep 23 16:47:07 2021 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Thu Sep 23 16:47:07 2021 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1541', remote='link-mtu 1557'
Thu Sep 23 16:47:07 2021 WARNING: 'cipher' is used inconsistently, local='cipher BF-CBC', remote='cipher AES-256-CBC'
Thu Sep 23 16:47:07 2021 WARNING: 'keysize' is used inconsistently, local='keysize 128', remote='keysize 256'
Thu Sep 23 16:47:07 2021 [server] Peer Connection Initiated with [AF_INET]server:1194
Thu Sep 23 16:47:08 2021 TUN/TAP device tun0 opened
Thu Sep 23 16:47:08 2021 /sbin/ip link set dev tun0 up mtu 1500
Thu Sep 23 16:47:08 2021 /sbin/ip addr add dev tun0 xxxxx/24 broadcast xxxxx
Thu Sep 23 16:47:08 2021 Initialization Sequence Completed

```

**Comment by dsommers** [comment]
`static-challenge` is currently not fully implemented properly, and the `openvpn2` user interface does have some known issues with 2FA in some setups.

Can you try to start the session using `openvpn3 session-start --config client.ovpn` ?

**Comment by telboon** [comment]
That was attempted as well, with the same results as above (2nd command ran on OpenVPN3)

```
$ openvpn3 session-start -c client.ovpn
Using configuration profile from file: client.ovpn
Session path: /net/openvpn/v3/sessions/bbc30e60s4a88s4f8bs88b4sfe7373e443df
Auth User name: user
Auth Password: 
session-start: ** ERROR ** Failed to connect: Connection, Client authentication failed: Authentication failed
```

**Comment by dsommers** [comment]
Full `static-challenge` support should be resolved with commit 3a73c2e6b007bf4c0837821550ec88345156562f, which will be part of the next v17 release.

**Comment by dsommers** [comment]
v17_beta has been released now.
